EIPS/eip-3675.md

```diff
@@ -0,0 +1,247 @@
+* Remove the existing PoW heaviest chain rule.
+* Adhere to the new PoS LMD-GHOST rule. 
+
+The new PoS LMD-GHOST fork choice rule is specified as follows. On each occurrence of a `POS_CHAINHEAD_SET` event including the first one, the following actions **MUST** be taken:
+* Consider the chain starting at genesis and ending with the block nominated by the event as the canonical blockchain.
+* Set the head of the canonical blockchain to the block nominated by the event.
+
+
+### Network
+
+The networking stack **SHOULD NOT** send the following messages if they advertise the descendant of any terminal PoW block:
+* [`NewBlockHashes (0x01)`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblockhashes-0x01)
+* [`NewBlock (0x07)`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblock-0x07)
+
+Beginning with the first `POS_BLOCK_FINALIZED` event, the networking stack **MUST** remove the handlers corresponding to the following ETH protocol messages:
+* [`NewBlockHashes (0x01)`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblockhashes-0x01)
+* [`NewBlock (0x07)`](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblock-0x07)
+
+Beginning with the first `POS_BLOCK_FINALIZED` event, peers that keep sending these messages **SHOULD** be disconnected.
+
+*Note:* The logic of message handlers that are not affected by this section **MUST** remain unchanged.
+
+
+## Rationale
+
+The changes specified in this EIP target a minimal requisite set of consensus and client software modifications to safely replace the existing proof-of-work consensus algorithm with the new proof-of-stake consensus represented by the already in-production beacon chain.
+
+This EIP was designed to minimize the complexity of hot-swapping the live consensus of the Ethereum network. Both the safety of the operation and time to production were taken into consideration. Additionally, a minimal changeset helps ensure that *most* smart contracts and services will continue to function as intended during and after the transition with little to no required intervention.
+
+### Total difficulty triggering the upgrade
+
+See [Security considerations](#terminal-total-difficulty-vs-block-number).
+
+### Halting the import of PoW blocks
+
+See [Security considerations](#halt-the-importing-of-pow-blocks).
+
+### Replacing block fields with constants
+
+Deprecated block fields are replaced with constant values to ensure the block format remains backwards compatible. Preserving the block format aids existing smart contracts and services in providing uninterrupted service during and after this transition.
+
+Particularly, this is important for those smart contracts that verify Merkle proofs of transaction/receipt inclusion and state by validating the hash of externally provided block header string against the corresponding value returned by the `BLOCKHASH` operation.
```

Comment:
```suggestion
Particularly, this is important for those smart contracts that verify Merkle proofs of transaction/receipt inclusion and state by validating the hash of externally provided block header against the corresponding value returned by the `BLOCKHASH` operation.
```
The `string` there I think makes the sentence a bit more confusing, as you may receive a block header in many forms such as a structured format, a byte array, a hex string, etc.

Reply:
You're right, theoretically they may receive a block header as a set of field values, then RLP encode it, compute a hash and verify this hash. Although it does seem like a suboptimal solution we better remove this "string".

Reply:
If you need to pull data out of the header, it may be easier to serialize from raw inputs than it is to parse/decerialize to pull bits out of the header.

Reply:
Arguably it may be cheaper given a string to verify it against the corresponding `BLOCKHASH` value by parsing out a `number` using some optimized RLP decoder that jumps between list elements and then read the needed field in the same fashion than RLP encoding the whole thing. Anyway, I am going to remove the "string" term.